nipype/interfaces/freesurfer/utils.py

```diff
@@ -1967,8 +1967,14 @@ def _format_arg(self, name, spec, value):
         if name == "lta_in" and self.inputs.invert_lta_in:
             spec = "--lta-inv %s"
         if name in ("fsl_out", "lta_out") and value is True:
-            value = self._list_outputs()[name]
-        return super(Tkregister2, self)._format_arg(name, spec, value)
+            if name == 'fsl_out':
+                name_actual = 'fsl_file'
+            if name == 'lta_out':
+                name_actual = 'lta_file'
+            value = self._list_outputs()[name_actual]
+            return super(Tkregister2, self)._format_arg(name_actual, spec, value)
+        else:
+            return super(Tkregister2, self)._format_arg(name, spec, value)
```

Comment:
I don't think this is quite right. You still want to pass `name` to `_format_arg`, you just need to get the correct output.

Does the following fix it for you?

```suggestion
            value = self._list_outputs()[f'{name[:3]}_file']
        return super()._format_arg(name, spec, value)
```

Reply:
Yes you are right. It seems that `name` for `_format_arg` only matters for logging.
For `name_actual`, I try to be explicit, but it is probably not important :)